### PR TITLE
Reformat dates to use a consistent format

### PR DESCRIPTION
**Pull request by dhawton**
Please provide a description for what this PR is for.

Continuing discussion from https://github.com/istio/istio.io/pull/12919#discussion_r1146431994

Reformatted dates in support status of Istio releases to use the same format. Formatting to a short month keeps things on a single line, except when (Expected) is added to the End of Life fields. Opted for short month names over yyyy-mm-dd as the it helps readability regardless of locale date formats, and maintains readability for individuals not used to reading that format.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Ambient
- [X] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: data/compatibility/supportStatus.yml [hunk 1]
```diff
@@ -9,20 +9,20 @@
   testedK8sVersions:
 - version: "1.17"
   supported: "Yes"
-  releaseDate: "February 14, 2023"
+  releaseDate: "Feb 14, 2023"
   eolDate: "~Sept 2023 (Expected)"
```
Comment (ericvn) [inline data/compatibility/supportStatus.yml:13]:
```suggestion
  eolDate: "~Sep 2023 (Expected)"
```

**Review by ericvn — COMMENTED** [review]
Looks like on that didn't get shortened to 3 characters.

File: data/compatibility/supportStatus.yml (hunk 1 above)
Comment (dhawton) [reply data/compatibility/supportStatus.yml:13]: Yup I missed that one. Thanks

**Comment by ericvn** [comment]
Looks good to me at https://deploy-preview-12954--preliminary-istio.netlify.app/latest/docs/releases/supported-releases/ both on my phone and computer.

**Comment by ericvn** [comment]
Adding a period does not seem to affect the flow (tried locally):
Screenshot 2023-03-27 at 1 23 04 PM

**Comment by ericvn** [comment]
Unfortunately, looking at the deploy/netlify output, a few of the lines are now split with the year on the following line. I'm wondering what happens if someone extends the `Release date` in layouts/shortcodes/support_status_table.html with a space. Does that get picked up and widen the table column?

**Comment by dhawton** [comment]
> Unfortunately, looking at the deploy/netlify output, a few of the lines are now split with the year on the following line. I'm wondering what happens if someone extends the `Release date` in layouts/shortcodes/support_status_table.html with a space. Does that get picked up and widen the table column?

What about leaving the periods off? It seems pretty common these days to not add the period with shortened month names. Might also be "better" than having a "hacky" space added to the column name. In my env without the periods, only 1.11's EOL date drops to another line (likely just because of font) or the `(Expected)` dates.

**Review by ericvn — APPROVED** [review]
Thanks for handling all the comments on this one.

**Pull request merged**